test/e2e/tests/block_test.go

```diff
@@ -6,6 +6,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
+	"github.com/cometbft/cometbft/internal/state"
```

Comment:
This (e2e tests and runner) can be considered external to CometBFT, so I don't think it's best to access "internal" from here. I'd propose to move `MedianTime` elsewhere in CometBFT (out of "internal"). What do you think?

Reply:
Totally. It could be moved to `types/block.go` or `types/validation.go`. The latter contains "abstract" methods receiving a `Commit` and a `ValidatorSet` as well.

Reply:
Thanks @cason for the input!
I moved it to `types/block.go`, and made it a method of `Commit`. Please take a look at the last commit

Reply:
In terms of readability, your solution is great :)